src/com/ms/silverking/numeric/StatSeries.java

```diff
@@ -9,238 +9,238 @@
+    } else {
+      return Double.NaN;
     }
-    
-    public static String summaryHeaderString() {
-        StringBuilder   sb;
-        
-        sb = new StringBuilder();
-        sb.append("mean");
-        sb.append('\t');
-        sb.append("median");
-        sb.append('\t');
-        sb.append("variance");
-        sb.append('\t');
-        sb.append("standardDeviation");
-        sb.append('\t');
-        sb.append("min");
-        sb.append('\t');
-        sb.append("max");
-        sb.append('\t');
-        sb.append("90%");
-        sb.append('\t');
-        sb.append("95%");
-        sb.append('\t');
-        sb.append("99%");
-        return sb.toString();
+  }
+
+  public double sum() {
+    return sum;
+  }
+
+  public double median() {
+    return median;
+  }
+
+  public double mean() {
+    return mean;
+  }
+
+  public double max() {
+    return max;
+  }
+
+  public double min() {
+    return min;
+  }
+
+  public double standardDeviation() {
+    return standardDeviation;
+  }
+
+  public double variance() {
+    return variance;
+  }
+
+  public String toString() {
+    StringBuilder sb;
+
+    sb = new StringBuilder();
+    sb.append(toSummaryString());
+    sb.append('\t');
+    for (double sample : samples) {
+      sb.append(sample);
+      sb.append('\t');
+    }
+    return sb.toString();
+  }
+
+  public String toSummaryString() {
+    StringBuilder sb;
+
+    sb = new StringBuilder();
+    sb.append(mean);
+    sb.append('\t');
+    sb.append(median);
+    sb.append('\t');
+    sb.append(variance);
+    sb.append('\t');
+    sb.append(standardDeviation);
+    sb.append('\t');
+    sb.append(min);
+    sb.append('\t');
+    sb.append(max);
+    sb.append('\t');
+    sb.append(percentile(90));
+    sb.append('\t');
+    sb.append(percentile(95));
+    sb.append('\t');
+    sb.append(percentile(99));
+    return sb.toString();
+  }
+
+  public String toSummaryStringLow() {
+    StringBuilder sb;
+
+    sb = new StringBuilder();
+    sb.append(mean);
+    sb.append('\t');
+    sb.append(median);
+    sb.append('\t');
+    sb.append(variance);
+    sb.append('\t');
+    sb.append(standardDeviation);
+    sb.append('\t');
+    sb.append(min);
+    sb.append('\t');
+    sb.append(max);
+    sb.append('\t');
+    sb.append(percentile(1));
+    sb.append('\t');
+    sb.append(percentile(5));
+    sb.append('\t');
+    sb.append(percentile(10));
+    return sb.toString();
+  }
+
+  public static String summaryHeaderString() {
+    StringBuilder sb;
+
+    sb = new StringBuilder();
+    sb.append("mean");
+    sb.append('\t');
+    sb.append("median");
+    sb.append('\t');
+    sb.append("variance");
+    sb.append('\t');
+    sb.append("standardDeviation");
+    sb.append('\t');
+    sb.append("min");
+    sb.append('\t');
+    sb.append("max");
+    sb.append('\t');
+    sb.append("90%");
+    sb.append('\t');
+    sb.append("95%");
+    sb.append('\t');
+    sb.append("99%");
+    return sb.toString();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 180:198, 204:222.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.